dids/diddht/bencode/bencode.go

```diff
@@ -0,0 +1,260 @@
+package bencode
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+)
+
+const (
+	DictionaryPrefix = 'd'
+	IntegerPrefix    = 'i'
+	ListPrefix       = 'l'
+	EndSuffix        = 'e'
+)
+
+// Marshal encodes the given input into a Bencode formatted byte array.
+// More information about Bencode can be found at:
+// https://wiki.theory.org/BitTorrentSpecification#Bencoding
+func Marshal(input any) ([]byte, error) {
+	switch v := input.(type) {
+	case string:
+		encoded := fmt.Sprintf("%d:%s", len(v), v)
```

Comment:
Is this length in bytes or in UTF-8 codepoints?

Reply:
according to the "spec", just string length

![image](https://github.com/TBD54566975/web5-go/assets/4887440/b7b18381-e3bd-43a1-9613-4966d7f522e6)

Reply:
Disturbingly ambiguous!

Reply:
Maybe the format doesn't support UTF-8 at all...